### PR TITLE
feat: Add migration to add `managed` column to credentials table

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

Ground work to support free AI credits. See context [here](https://www.notion.so/n8n/Free-AI-credits-1565b6e0c94f8074b254da8db0fb6219?pvs=4).

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-2995/[n8n-backend]-add-new-column-to-the-credentials-table-to-differentiate

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/cli/src/databases/entities/credentials-entity.ts [hunk 1]
```diff
@@ -29,6 +29,9 @@ export class CredentialsEntity extends WithTimestampsAndStringId implements ICre
 	@OneToMany('SharedCredentials', 'credentials')
 	shared: SharedCredentials[];
 
+	@Column({ default: false })
+	managed: boolean;
```
Comment (tomi) [inline packages/cli/src/databases/entities/credentials-entity.ts:33]: I'm wondering if a `type` column would make more sense with possible values of `"default" | "managed"` 🤔 I don't see us adding a third one any time soon tho. So I'm fine with this or `type`. Your call. If we keep it as a boolean, we could name it `isManaged` to be explicit that it's a boolean.

Also would be nice to have a comment here explaining what it means that a credential is "managed"

Comment (RicardoE105) [reply packages/cli/src/databases/entities/credentials-entity.ts:33]: My original idea was also call it `type`, but I do not see a future where we need another type 🤔. Let's call it `isManaged` and add the comment.

File: packages/cli/src/databases/entities/credentials-entity.ts [hunk 2]
```diff
@@ -29,6 +29,15 @@ export class CredentialsEntity extends WithTimestampsAndStringId implements ICre
 	@OneToMany('SharedCredentials', 'credentials')
 	shared: SharedCredentials[];
 
+	/**
+	 * Whether the credential is managed by n8n. We currently use this flag
+	 * to provide OpenAI free credits on cloud. Managed credentials cannot be
+	 * edited by the user.
+	 */
+
```
Comment (tomi) [inline packages/cli/src/databases/entities/credentials-entity.ts:37]:
```suggestion
```

**Review by tomi — DISMISSED** [review]

**Comment by RicardoE105** [comment]
@tomi thanks for syncing with master 🙏

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.74.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.1)

**Comment by janober** [comment]
Got released with [`n8n@1.74.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.0)

**Pull request merged**